### PR TITLE
Fix flakiness with the CI build

### PR DESCRIPTION
**Pull request by jrhee17**
Motivation:

CI builds are currently flaky due to issues other then test failure

### Javadoc generation failure

```
* What went wrong:
Execution failed for task ':examples:tutorials:rest-api-annotated-service:javadoc'.
> Javadoc generation failed. Generated Javadoc options file (useful for troubleshooting): '/home/ec2-user/actions-runner/_work/armeria/armeria/examples/tutorials/rest-api-annotated-service/build/tmp/javadoc/javadoc.options'
```

I believe the reason for this failure is due to the javadoc command trying to fetch linked assets for `java se`.
The [bug](https://bugs.java.com/bugdatabase/view_bug.do?bug_id=8212233) that was mentioned previously has been fixed since 11.03, so I think we can use offline links again.
Additionally, we may also upload the `javadoc.options` file on build failure to get a better idea of what went wrong in the future.

Screen Shot 2023-02-15 at 9 54 10 AM

### Insufficient heap

I wrongly assumed that forked tasks weren't allocated enough memory.
Rather, it seems like the whole system didn't have enough memory due to proguard's memory leak.
After updating to the most recent proguard version, it seems like this [issue](https://github.com/Guardsquare/proguard/issues/283) doesn't persist. I've also reverted the max heap per fork since there seems to be no issue.

### Stuck builds while trying to retrieve a thread dump

For an unknown reason, it seems like windows systems fail to generate a thread dump for `KotlinCompileDaemon`.
I think we're not interested in daemon processes thread stacks anyways, so I've removed processes containing `Daemon` from the target for thread dumps.

Screen Shot 2023-02-14 at 10 12 30 PM

Modifications:

- Use javadoc offline links for java se
- Upload `javadoc.options` files on build failure
- Upgraded proguard and reverted max heap for forked processes
- Filter out daemons when taking thread dumps

Result:

- More stable builds

### CODE REVIEW

**Comment by minwoox** [comment]
Let us know when this is ready. 😄

**Comment by jrhee17** [comment]
Ready for review 🙏

**Comment by ikhoon** [comment]
> After updating to the most recent proguard version, it seems like this https://github.com/Guardsquare/proguard/issues/283 doesn't persist. I've also reverted the max heap per fork since there seems to be no issue.

Happy to hear that. 🥳 The fix seems to use more heap but it is released quickly.
https://github.com/Guardsquare/proguard-core/pull/73#issuecomment-1275704352

**Review by ikhoon — APPROVED** [review]
💯

**Review by minwoox — APPROVED** [review]
Thanks! 👍 👍 👍

File: site/src/pages/community/developer-guide.mdx [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ## Build requirements
 
-- [OpenJDK 11-19](https://adoptium.net/)
+- [OpenJDK 17-19](https://adoptium.net/)
```
Comment (jrhee17) [inline site/src/pages/community/developer-guide.mdx:5]:
```suggestion
- [OpenJDK 11-19](https://adoptium.net/)
```

**Review by trustin — APPROVED** [review]

**Pull request merged**